### PR TITLE
feat(CustomParams): Added support for custom query parameters

### LINKED ISSUE #70: add ability to supply custom query parameters
we have a use case where we want to supply custom query parameters.

route looks like:

http://localhost:61868/route/v1/Reports?effectiveDate=2018-06-30

it would be great if we could supply Param like so.

```
service.Query().Param('effectiveDate', '2018-06-30').Exec()
```

also, thanks for getting the GetUrl functionality out so quick!

### PR DESCRIPTION
**Pull request by axeltaylor**
Adds support to provide custom query parameters.
Also closes #70

### CODE REVIEW

**Comment by albert-asin** [comment]
I need this new feature in my project
It looks great and could solve my problem
thanks for the idea

**Comment by axeltaylor** [comment]
> I need this new feature in my project
> It looks great and could solve my problem
> thanks for the idea

@StefH what do you think?

**Review by StefH — APPROVED** [review]

**Comment by StefH** [comment]
the code + unit tests look fine

I'm not sure if this is 100% odata compatible, but if it helps you, I will publish a new version this week

**Comment by axeltaylor** [comment]
Thank you @StefH , regarding the odata compatibility I found this information in the specification:  http://docs.oasis-open.org/odata/odata/v4.01/cs01/part2-url-conventions/odata-v4.01-cs01-part2-url-conventions.html#sec_CustomQueryOptions

It seems that is allowed and compatible but maybe we should add logic to not allow parameters starting with $ or @ characters. 
> Custom query options MUST NOT begin with a $ or @ character.

What do you think?

**Comment by StefH** [comment]
Good investigation.

I think it's good to verify the parameters for $ and @, and throw exception when the value is invalid.

File: src/angularODataQuery.ts [hunk 1]
```diff
@@ -258,4 +273,11 @@ export class ODataQuery<T> extends ODataOperation<T> {
     private extractArrayDataWithCount(res: HttpResponse<IODataResponseModel<T>>, config: ODataConfiguration): ODataPagedResult<T> {
         return config.extractQueryResultDataWithCount(res);
     }
+
+    private checkReservedCustomQueryOptionKey(key: string): string {
+        if (key.indexOf('$') === 0 || key.indexOf('@') === 0) {
```
Comment (StefH) [inline src/angularODataQuery.ts:281]: I think this method cannot handle undefined/null `key` ? Maybe add a null/undefined check?

**Review by StefH — CHANGES_REQUESTED** [review]
Looks good.
Please take a look at my comment; I'm not sure if this applies..

**Comment by illustriousmuz** [comment]
Hi, this would be a very useful feature for a project I'm currently working on. Any ETA for a release? Thanks

**Review by StefH — APPROVED** [review]
Thank you

**Comment by StefH** [comment]
@albert-asin and @illustriousmuz  : new version will be available in some hours

**Pull request merged**